src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,227 @@
+                addNewRelationCarsDrivers(car.getId(), driver.getId()));
+        return car;
+    }
+
+    private void deleteRelationsCarsDrivers(Long carId) {
+        String deleteRelationsRequest = "DELETE FROM cars_drivers WHERE car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteRelationsStatement
+                        = connection.prepareStatement(deleteRelationsRequest)) {
+            deleteRelationsStatement.setLong(1, carId);
+            deleteRelationsStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Couldn't delete relations in cars_drivers "
+                    + "for car by id: " + carId, e);
+        }
+    }
+
+    private void addNewRelationCarsDrivers(Long carId, Long driverId) {
+        String addRelationRequest = "INSERT INTO cars_drivers (car_id, driver_id) "
+                + "VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement addRelationStatement
+                        = connection.prepareStatement(addRelationRequest)) {
+            addRelationStatement.setLong(1, carId);
+            addRelationStatement.setLong(2, driverId);
+            addRelationStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Couldn't add relations in cars_drivers "
+                    + "between car by id: " + carId
+                    + " and driver by id: " + driverId, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement
+                        = connection.prepareStatement(deleteRequest)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new RuntimeException("Couldn't delete car by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverRequest = "SELECT id "
+                + "FROM cars c "
+                + "JOIN cars_drivers cd "
+                + "ON c.id = cd.car_id "
+                + "WHERE driver_id = ?";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement
+                        = connection.prepareStatement(getAllByDriverRequest)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                Long carId = resultSet.getObject("id", Long.class);
+                cars.add(get(carId).get());
```

Comment:
you shouldn't use the get method here, you should write a query that will receive all needed information, also for getting drivers for a car you need to call method below